pyanaconda/ui/tui/spokes/root_password.py

```diff
@@ -70,8 +78,15 @@ def showable(self):
 
     @property
     def mandatory(self):
-        """Only mandatory if no admin user has been requested."""
-        return not self._users_module.CheckAdminUserExists()
+        """Only mandatory if no admin user has been requested.
+
+        If root_password_must_be_set is set to True in the config file,
+        root password must be set, regardless of admin user existing or not.
+        """
+        if conf.security.root_password_must_be_set:
+            return not self._users_module.IsRootPasswordSet
+        else:
+            return not self._users_module.CheckAdminUserExists()
```

Comment:
The checks could be moved to new functions in `pyanaconda.ui.lib.users`.

Reply:
That's actually a really good idea, lots can be deduplicated this way, let's do that! :)

BTW, I think we should do this upstream in some form regardless of if the option ends up upstream as well.